### PR TITLE
Don't use the The Word Which Must Not Be Used

### PR DESCRIPTION
**Pull request by rouault**
I limited the renaming to things that are not in the public API and are not setting names. This is probably more than we strictly need to do, but at least this will help doing potential future similar 'grep' exercises.

### CODE REVIEW

**Review by nyalldawson — APPROVED** [review]

**Comment by nyalldawson** [comment]
@rouault can you approve https://github.com/qgis/QGIS-Documentation/pull/9296 too?

>  What is the procedure to make spell_check happy?

This particular misspell has been a long-standing issue, which isn't easy/possible to fix before 4.0. Just ignore it, as it will trip every time the layout exporter header is modified...

**Comment by nyalldawson** [comment]
working on a backport now...

File: python/PyQt6/core/auto_additions/qgslayoutexporter.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
...
...o effect.', 'useOgcBestPracticeFormatGeoreferencing': '``True`` if OGC "best practice" format georeferencing should be used.\n\n.. warning::\n\n   This results in geospatial PDF files compatible with a unnamed suite of tools starting with Terra and ending with Go, but\n   can break compatibility with the built-in Acrobat geospatial tools (yes, Geospatial PDF\n   format is a mess!).\n\nIf PdfExportSettings.writeGeoPdf is ``False`` than this option has no effect.', 'includeGeoPdfFeatures': '``True`` if feature vector information (such as attributes) should be exported during Geospatial PDF exports.\n\nIf PdfExportSettings.writeGeoPdf is ``False`` than this option has no effect.', 'exportThemes': "Optional list of map themes to export as Geospatial PDF layer groups.\n\nIf set, map item's which are not assigned a specific map theme will iterate through all listed\nthemes and a Geospatial PDF layer group will be created for each.\n\nIf PdfExportSettings.writeGeoPdf is ``False`` than this option has no effect.", 'predefinedMapScales': 'A list of predefined scales to use with the layout. This is used\nfor maps which are set to the predefined atlas scaling mode.\n\n.. versionadded:: 3.10'}
```
Comment (matt-needle) [inline python/PyQt6/core/auto_additions/qgslayoutexporter.py:22]: Sometimes the G of Geospatial is capitalised, but oftentimes not.

File: python/PyQt6/core/auto_generated/providers/qgsprovidersublayerdetails.sip.in [hunk 2]
```diff
@@ -21,7 +21,7 @@ according to a specific data provider.
 
    It is possible that multiple data providers may be able to utilize the same underlying
    sub layer in a single dataset, yet will interpret this layer differently. For instance, a layer
-   in a GeoPDF document can be interpreted as either a vector layer by the OGR data provider or a raster
+   in a Geospatial PDF document can be interpreted as either a vector layer by the OGR data provider or a raster
```
Comment (matt-needle) [inline python/PyQt6/core/auto_generated/providers/qgsprovidersublayerdetails.sip.in:24]: In this instance, the G is capitalised whereas in most others, it is not.

**Pull request merged**